rows.go

```diff
@@ -229,30 +231,23 @@ func (scd *snowflakeChunkDownloader) Next() ([]*string, error) {
 		if scd.CurrentChunkIndex > 1 {
 			scd.Chunks[scd.CurrentChunkIndex-1] = nil // detach the previously used chunk
 		}
-		scd.ChunksMutex.Unlock()
-
-		ticker := time.NewTicker(time.Second)
-		for range ticker.C {
-			scd.ChunksMutex.Lock()
+		for scd.Chunks[scd.CurrentChunkIndex] == nil {
+			glog.V(2).Infof("waiting for chunk idx: %v/%v",
+				scd.CurrentChunkIndex+1, len(scd.ChunkMetas))
 			err := scd.checkErrorRetry()
 			if err != nil {
 				scd.ChunksMutex.Unlock()
-				ticker.Stop()
 				return nil, err
 			}
-			glog.V(2).Infof("waiting for chunk idx: %v/%v",
-				scd.CurrentChunkIndex+1, len(scd.ChunkMetas))
-			scd.CurrentChunk = scd.Chunks[scd.CurrentChunkIndex]
-			scd.ChunksMutex.Unlock()
-			if scd.CurrentChunk != nil {
-				ticker.Stop()
-				// kick off the next download
-				glog.V(2).Infof("ready: chunk %v", scd.CurrentChunkIndex)
-				scd.CurrentChunkSize = len(scd.CurrentChunk)
-				scd.schedule()
-				break
-			}
+			scd.DoneDownloadCond.Wait()
```

Comment:
Wait for the downloader to broadcast the event, 1) one download finishes or 2) an error occurs.

Reply:
sure, will do.